File: backend/ResearchCruiseApp/Application/Models/DTOs/Cruises/CruiseDto.cs
namespace ResearchCruiseApp.Application.Models.DTOs.Cruises;


public class CruiseDto
{
    public Guid Id { get; set; }
    
    public string Number { get; set; } = null!;

    public string StartDate { get; set; } = null!;

    public string EndDate { get; set; } = null!;
    
    public Guid MainCruiseManagerId { get; set; }
    
    public string MainCruiseManagerFirstName { get; set; } = null!;
    
    public string MainCruiseManagerLastName { get; set; } = null!;
    
    public Guid MainDeputyManagerId { get; set; }
    
    public List<CruiseApplicationShortInfoDto> CruiseApplicationsShortInfo { get; set; } = [];

    public string Status { get; init; } = null!;
}   

File: frontend/src/modules/cruise-applications/models/CruiseDto.ts
import { CruiseApplicationShortInfoDto } from '@/cruise-applications/models/CruiseApplicationShortInfoDto';

export type CruiseDto = {
  id: string;
  number: string;
  startDate: string;
  endDate: string;
  mainCruiseManagerId: string;
  mainCruiseManagerFirstName: string;
  mainCruiseManagerLastName: string;
  mainDeputyManagerId: string;
  cruiseApplicationsShortInfo: CruiseApplicationShortInfoDto[];
  status: 'Nowy' | 'Potwierdzony' | 'Zakończony';
};
